From 569657a33785bf1641b5150bc273af6ad79a2125 Mon Sep 17 00:00:00 2001
From: Fei Wang <fei.w.wang@intel.com>
Date: Thu, 14 Jan 2021 13:15:32 -0500
Subject: [PATCH 45/73] lavc/vaapi_hevc/vp9: support 444 8/10bit enc

hevc 444 8bit:
ffmpeg -hwaccel vaapi -vaapi_device /dev/dri/renderD128 -v verbose -f rawvideo -pix_fmt 0yuv -s:v 1920x1080 -i 1920x1080_100.ayuv -vf 'format=0yuv,hwupload' -c:v hevc_vaapi -profile:v rext -rc_mode CQP -g 30 -qp 28 -slices 1 -low_power 1 -y out.h265

hevc 444 10bit:
ffmpeg -hwaccel vaapi -vaapi_device /dev/dri/renderD128 -v verbose -f rawvideo -pix_fmt y410 -s:v 1920x1080 -i 1920x1080_100.y410 -vf 'format=y410,hwupload' -c:v hevc_vaapi -profile:v rext -rc_mode CQP -g 30 -slices 1 -bf 3  -low_power 1 -y out.h265

vp9 444 8bit:
ffmpeg -hwaccel vaapi -vaapi_device /dev/dri/renderD128 -v verbose -f rawvideo -pix_fmt 0yuv -s:v 1920x1080 -i 1920x1080_100.ayuv -vf 'format=0yuv,hwupload' -c:v vp9_vaapi -rc_mode CQP -g 30 -qp 28 -slices 1 -low_power 1 -y out.iv

vp9 444 10bit:
ffmpeg -hwaccel vaapi -vaapi_device /dev/dri/renderD128 -v verbose -f rawvideo -pix_fmt y410 -s:v 1920x1080 -i 1920x1080_100.y410 -vf 'format=y410,hwupload' -c:v vp9_vaapi -rc_mode CQP -g 30 -qp 28 -slices 1 -low_power 1 -y out.ivf

Signed-off-by: Fei Wang <fei.w.wang@intel.com>
---
 libavcodec/vaapi_encode_h265.c | 2 ++
 libavcodec/vaapi_encode_vp9.c  | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/libavcodec/vaapi_encode_h265.c b/libavcodec/vaapi_encode_h265.c
index f478025e42..bf239987f3 100644
--- a/libavcodec/vaapi_encode_h265.c
+++ b/libavcodec/vaapi_encode_h265.c
@@ -1310,6 +1310,8 @@ static const VAAPIEncodeProfile vaapi_encode_h265_profiles[] = {
 #if VA_CHECK_VERSION(1, 2, 0)
     { FF_PROFILE_HEVC_REXT,     8, 3, 1, 0, VAProfileHEVCMain422_10 },
     { FF_PROFILE_HEVC_REXT,    10, 3, 1, 0, VAProfileHEVCMain422_10 },
+    { FF_PROFILE_HEVC_REXT,     8, 4, 0, 0, VAProfileHEVCMain444    },
+    { FF_PROFILE_HEVC_REXT,    10, 3, 0, 0, VAProfileHEVCMain444_10 },
     { FF_PROFILE_HEVC_REXT,    12, 3, 1, 1, VAProfileHEVCMain12     },
     { FF_PROFILE_HEVC_REXT,    12, 3, 1, 0, VAProfileHEVCMain422_12 },
     { FF_PROFILE_HEVC_REXT,    12, 3, 0, 0, VAProfileHEVCMain444_12 },
diff --git a/libavcodec/vaapi_encode_vp9.c b/libavcodec/vaapi_encode_vp9.c
index 892ad770c6..a6a5b8ef0b 100644
--- a/libavcodec/vaapi_encode_vp9.c
+++ b/libavcodec/vaapi_encode_vp9.c
@@ -228,7 +228,9 @@ static av_cold int vaapi_encode_vp9_configure(AVCodecContext *avctx)
 
 static const VAAPIEncodeProfile vaapi_encode_vp9_profiles[] = {
     { FF_PROFILE_VP9_0,  8, 3, 1, 1, VAProfileVP9Profile0 },
+    { FF_PROFILE_VP9_1,  8, 4, 0, 0, VAProfileVP9Profile1 },
     { FF_PROFILE_VP9_2, 10, 3, 1, 1, VAProfileVP9Profile2 },
+    { FF_PROFILE_VP9_3, 10, 3, 0, 0, VAProfileVP9Profile3 },
     { FF_PROFILE_UNKNOWN }
 };
 
-- 
2.17.1

